metals/src/main/scala/scala/meta/internal/metals/codeactions/ActionableDiagnostic.scala

```diff
@@ -0,0 +1,65 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals._
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+class ActionableDiagnostic() extends CodeAction {
+  override def kind: String = l.CodeActionKind.QuickFix
+
+  override def contribute(
+      params: l.CodeActionParams,
+      token: CancelToken,
+  )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = {
+
+    def path =
+      params.getTextDocument.getUri.toAbsolutePath.toURI.toString
+
+    def createDiagnosticAction(
+        diagnostic: l.Diagnostic,
+        relatedInformation: l.DiagnosticRelatedInformation,
+    ): l.CodeAction = {
+      val codeAction = new l.CodeAction()
+      val to = relatedInformation.getMessage
+      codeAction.setKind(l.CodeActionKind.QuickFix)
+      codeAction.setTitle(ActionableDiagnostic.title)
+      codeAction.setDiagnostics(List(diagnostic).asJava)
+      val range = relatedInformation.getLocation.getRange
+      val textEdit = new l.TextEdit(range, to)
+      codeAction.setEdit(
+        new l.WorkspaceEdit(
+          Map(path -> List(textEdit).asJava).asJava
+        )
+      )
+      codeAction
+    }
+
+    val codeActions = params
+      .getContext()
+      .getDiagnostics()
+      .asScala
+      .groupBy {
+        case ScalacDiagnostic.ScalaAction(relatedInformation) =>
+          Some(relatedInformation)
+        case _ => None
+      }
+      .collect {
+        case (Some(relatedInformation), diags)
+            if params.getRange().overlapsWith(diags.head.getRange()) =>
+          createDiagnosticAction(diags.head, relatedInformation)
+      }
+      .toList
+      .sorted
+
+    Future.successful(codeActions)
+  }
+}
+
+object ActionableDiagnostic {
+  def title = "Apply suggestion from build server"
```

Comment:
```suggestion
  def title = "Apply suggestion"
```
I don't think the user even needs to know that this is "from the build server".